packages/ipfs-core/src/interface/basic.ts

```diff
@@ -0,0 +1,28 @@
+import CID from 'cids'
```

Comment:
Why are there `.ts` files being added?  The agreed approach for this project is to generate types from JSDoc comments.

Reply:
I think we talked about it. This are interface definitions for things we claim users can provide own implementations for. I think it would make sense to migrate some e.g. to https://github.com/ipfs/interface-datastore/ and others might need a new home. However it made most sense to me to put them here first and migrate them elsewhere afterwards.

This makes it possible to specify what interface option like bitswap, blockService, ipld, repo etc have to implement to qualify. Furthermore this dramatically increased type coverage. I think what we'd need to do in ipfs-bitswap, ipfs-block-service etc... is say that they `@implements {BitSwap}` etc... That way

1. We code against the interface not against our own implementation. We already use some internal properties that aren't part of the interface which implies that interface we have has gaps and we should probably rethink interface.

2. Other implementations could be coded against interface and `@implements {X}` will enforce compatibility, as opposed to doing it against markdown documents that are (, as it turns out) out of sync.

So I don't think it's an alternative to generating types from code. It's complementary and forces us to uphold interface contract we our selfs defined.